libs/geometry/include/geometry/TangentSpaceMesh.h

```diff
@@ -170,17 +180,47 @@ class TangentSpaceMesh {
          */
         Builder& uvs(filament::math::float2 const* uvs, size_t stride = 0) noexcept;
 
+        /**
+         * Sets "auxiliary" attributes that will be properly mapped when remeshed.
+         *
+         * @param attribute The attribute of the data to be stored
+         * @param data The data to be store
+         * @param stride The stride for iterating through `attribute`
+         * @return Builder
+         */
+        Builder& aux(AuxAttribute attribute, InData data, size_t stride = 0) noexcept;
```

Comment:
What if I mismatch the auxiliary attribute type with a pointer to the wrong data type?

```
filament::math::float3 uvs;
...
TangentSpaceMesh::Builder()
    .aux(AuxAttribute::UV1, &uvs)
```

I think this will still compile and run without error, but inside `TangetSpaceMeshOutput::passthrough`, we assume a different type (UV1 should be `float2`).

Reply:
Good point.  I added an assert in `build()` to validate the input type of the attributes.